### PR TITLE
[bitnami/gitea] Add new customizable environment variables - OAUTH2_CLIENT

### PR DESCRIPTION
**Pull request by ikamii**
[bitnami/gitea] Add new customizable environment variables

                    [outh2_client]
                    1. ENABLE_AUTO_REGISTRATION (default: false)
                    2. USERNAME (default: nickname)

### Description of the change

I have made changes in scripts `gitea-env.sh`, `libgitea.sh` and in `app.ini.template to add two new customizable environment variables to configure oauth2_client.

README.md has been updated with these changes.

### Benefits

Users can enable `Auto Registration` while logging in via OIDC  and change username claim as well

### Applicable issues

There are no open issues for these changes.
### Additional information

I would also make changes in `bitnami/charts/gitea` to enable same features for the helm chart.

### CODE REVIEW

File: bitnami/gitea/1/debian-12/rootfs/opt/bitnami/scripts/libgitea.sh [hunk 1]
```diff
@@ -221,6 +221,9 @@ gitea_update_conf_file() {
     is_empty_value "$GITEA_SMTP_USER" || gitea_conf_set "mailer" "USER" "$GITEA_SMTP_USER"
     is_empty_value "$GITEA_SMTP_PASSWORD" || gitea_conf_set "mailer" "PASSWD" "$GITEA_SMTP_PASSWORD"
     is_empty_value "$GITEA_LFS_ROOT_PATH" || gitea_conf_set "lfs" "PATH" "$GITEA_LFS_ROOT_PATH"
+
+    is_empty_value "$GITEA_OAUTH2_CLIENT_AUTO_REGISTRATION" || gitea_conf_set "oauth2_client" "ENABLE_AUTO_REGISTRATION" "$GITEA_OAUTH2_CLIENT_AUTO_REGISTRATION"
+    is_empty_value "$GITEA_OAUTH2_CLIENT_USERNAME" || gitea_conf_set "oauth2_client" "USERNAME" "$GITEA_OAUTH2_CLIENT_USERNAME"
```
Comment (andresbono) [inline bitnami/gitea/1/debian-12/rootfs/opt/bitnami/scripts/libgitea.sh:226]: I wonder if we should add some input validation for these 2 new env-vars in `gitea_validate`. Something like:

```bash
    check_true_false_value() {
        if ! is_true_false_value "${!1}"; then
            print_validation_error "The allowed values for $1 are [true, false]"
        fi
    }
    check_multi_value() {
        if [[ " ${2} " != *" ${!1} "* ]]; then
            print_validation_error "The allowed values for ${1} are: ${2}"
        fi
    }


    check_true_false_value 'GITEA_OAUTH2_CLIENT_AUTO_REGISTRATION_ENABLED'
    check_multi_value 'GITEA_OAUTH2_CLIENT_USERNAME' 'userid nickname preferred_username email'
```

File: bitnami/gitea/1/debian-12/rootfs/opt/bitnami/scripts/gitea-env.sh [hunk 2]
```diff
@@ -60,6 +60,8 @@ gitea_env_vars=(
     GITEA_SMTP_FROM
     GITEA_SMTP_USER
     GITEA_SMTP_PASSWORD
+    GITEA_OAUTH2_CLIENT_AUTO_REGISTRATION
```
Comment (andresbono) [inline bitnami/gitea/1/debian-12/rootfs/opt/bitnami/scripts/gitea-env.sh:63]: What if we call it `GITEA_OAUTH2_CLIENT_AUTO_REGISTRATION_ENABLED`?

**Review by andresbono — COMMENTED** [review]
Thank you so much for your contribution! Could you please check my comments?

File: bitnami/gitea/1/debian-12/rootfs/opt/bitnami/scripts/gitea-env.sh (hunk 2 above)
Comment (ikamii) [reply bitnami/gitea/1/debian-12/rootfs/opt/bitnami/scripts/gitea-env.sh:63]: Sure!

File: bitnami/gitea/1/debian-12/rootfs/opt/bitnami/scripts/libgitea.sh (hunk 1 above)
Comment (ikamii) [reply bitnami/gitea/1/debian-12/rootfs/opt/bitnami/scripts/libgitea.sh:226]: Of course if you think we should. Should I do it or are you going to update it?

**Review by andresbono — APPROVED** [review]
LGTM. Thank you for your contribution! 🎉

**Pull request merged**